include/AdePT/core/AsyncAdePTTransport.cuh

```
@@ -789,20 +797,37 @@ void TransportLoop(int trackCapacity, int scoringCapacity, int numThreads, Track
       // *** Particle injection ***
       // --------------------------
       if (gpuState.injectState == InjectState::Idle) {
-        for (auto &eventState : eventStates) {
-          if (const auto state = eventState.load(std::memory_order_acquire); state == EventState::G4RequestsFlush) {
-            eventState = EventState::Inject;
+        for (int i = 0; i < numThreads; ++i) {
+          auto &eventState = eventStates[i];
+          const auto state = eventState.load(std::memory_order_acquire);
+
+          // HOTFIX:
+          // the current problem is that the injected particle takes too long before it shows up
+          // until then, the EventState has counted up and requested a Flush of the event, before the particle was
```

Due to
https://github.com/apt-sim/AdePT/pull/367/files#diff-e7786b5f8e89172d219a16625b438e59af0580796e9f7c19de3a946cf65e8085R853
The CPU thread blocks until the injection is complete, so I don't think this counter has any effect (Since we only start counting **after** the copy has already finished) The delay is probably hiding the error somewhere else